sagemathinc/cocalc
# Ts r misc

# Description
Todo:
- [x] Testing for NumberInput
- [x] Testing for TextInput

Typescript
 - SimpleX
 - SkinnyError
 - ErrorDisplay
 - LabeledRow
 - NumberInput
 - TextInput
 - Spinner

Added some render tests and some interaction tests.

# Testing Steps
1. Check loading screens for our spinner
1. Try to use a bogus coupon (SimpleX, SkinnyError, ErrorDisplay)
1. View project settings (LabeledRow)
1. Change editor tab/fontsize/autosave settings (NumberInput)
1. Edit project description and title (TextInput) 

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
